src/AzureAuth/Commands/Ado/CommandPat.cs

```diff
@@ -60,19 +111,161 @@ private enum OutputMode
         [Option(ScopeOption, ScopeHelp, CommandOptionType.MultipleValue)]
         private string[] Scopes { get; set; } = null;
 
+        [Option(PromptHintOption, PromptHintHelp, CommandOptionType.SingleValue)]
+        private string PromptHint { get; set; } = null;
+
         [Option(OutputOption, OutputHelp, CommandOptionType.SingleValue)]
         private OutputMode Output { get; set; } = OutputMode.Token;
 
+        [Option(ModeOption, ModeHelp, CommandOptionType.MultipleValue)]
+        private IEnumerable<AuthMode> AuthModes { get; set; } = new[] { AuthMode.Default };
+
+        [Option(DomainOption, DomainHelp, CommandOptionType.SingleValue)]
+        private string Domain { get; set; } = AzureAuth.Ado.Constants.PreferredDomain;
+
         /// <summary>
         /// Executes the command and returns a status code indicating the success or failure of the execution.
         /// </summary>
         /// <param name="logger">The <see cref="ILogger{T}"/> instance that is used for logging.</param>
+        /// <param name="publicClientAuth">An <see cref="IPublicClientAuth"/>.</param>
         /// <param name="eventData">Lasso injected command event data.</param>
         /// <returns>An integer status code. 0 for success and non-zero for failure.</returns>
-        public int OnExecute(ILogger<CommandPat> logger, CommandExecuteEventData eventData)
+        public int OnExecute(ILogger<CommandPat> logger, IPublicClientAuth publicClientAuth, CommandExecuteEventData eventData)
         {
-            logger.LogInformation("coming soon");
+            if (!this.ValidOptions(logger))
+            {
+                return 1;
+            }
+
+            var accessToken = this.AccessToken(publicClientAuth, eventData);
+            if (accessToken == null)
+            {
+                logger.LogError("Failed to acquire an Azure DevOps access token. Re-run with '--verbosity debug' for more info.");
+                return 1;
+            }
+
+            IPatCache cache = null;
+            try
+            {
+                cache = this.Cache();
+            }
+            catch (MsalCachePersistenceException e)
+            {
+                logger.LogError($"Failed to validate cache persistence: {e.Message}");
+                return 1;
+            }

```

Comment:
I'd prefer to see this catch closer to the potential throw, explicitly around the cache check in that method so it's clearer what can do the throw.

Reply:
I went ahead and removed the persistence check for now and I'll come back to it in a follow up PR to improve all error handling for this subcommand.